src/scheduler/buckets.c

```diff
@@ -1100,16 +1098,28 @@ check_node_buckets(status *policy, server_info *sinfo, queue_info *qinfo, resour
 	if (nodepart != NULL) {
 		int i;
 		int can_run = 0;
+		static schd_error *failerr = NULL;
+		if (failerr == NULL) {
+			failerr = new_schd_error();
+			if (failerr == NULL)
+				return NULL;
+		} else
+			clear_schd_error(failerr);
+
 		for (i = 0; nodepart[i] != NULL; i++) {
 			nspec **nspecs;
 			snprintf(log_buffer, MAX_LOG_SIZE, "Evaluating placement set: %s", nodepart[i]->name);
 			schdlog(PBSEVENT_DEBUG3, PBS_EVENTCLASS_JOB, LOG_DEBUG, resresv->name, log_buffer);
 
+			clear_schd_error(err);
 			nspecs = map_buckets(policy, nodepart[i]->bkts, resresv, err);
 			if (nspecs != NULL)
 				return nspecs;
-			if (err->status_code == NOT_RUN)
+			if (err->status_code == NOT_RUN) {
+				if (failerr->status_code == SCHD_UNKWN)
+					move_schd_error(failerr, err);
 				can_run = 1;
```

Comment:
i know it's not your code, but do you know why 'can_run' is being set to 1 if err status code is NOT_RUN ?

Reply:
NOT_RUN means the job can possibly run but not right now.  NEVER_RUN means the job can never run.  The variable is saying the job can possibly run.

Reply:
Thanks for clarifying Bhroam